### PR TITLE
Suggested refactor

### PR DESCRIPTION
**Pull request by butlersrepos**
This is more opinionated than suggesting a "better practice".
@aaronmalone

### CODE REVIEW

File: src/main/csv.js [hunk 1]
```diff
@@ -1,31 +1,42 @@
-const _ = require('lodash');
-
-const initial = _.initial;
-const last = _.last;
-const trim = _.trim;
+const initial = require('lodash').initial;
```
Comment (butlersrepos) [inline src/main/csv.js]: Multiple `require` calls are instantaneous, the first one caches the module like a singleton.

File: src/main/csv.js [hunk 2]
```diff
@@ -1,31 +1,42 @@
...
+const last = require('lodash').last;
+const trim = require('lodash').trim;
 
 const DOUBLE_QUOTE = '"';
 const COMMA = ',';
 
+function isSeparatorComma(char, quoted) {
```
Comment (butlersrepos) [inline src/main/csv.js]: "functional" means LOTS of functions right? Am I doing functional moar?

File: src/main/csv.js [hunk 3]
```diff
@@ -1,31 +1,42 @@
...
+		}
+
+		const char = input.charAt(index++);
+
+		quoted = char === DOUBLE_QUOTE ? !quoted : quoted;
+		if (char !== DOUBLE_QUOTE) {
+			parts = isSeparatorComma(char, quoted) ? startNewBlankItem(parts) : addToLastItem(parts, char);
+		}
+
+		return splitRecursively(parts, index, quoted);
```
Comment (butlersrepos) [inline src/main/csv.js]: I like to limit my `return`s to 2 generally speaking. The first being an escape, fail-fast. The second being the expected flow. Along the way we just create the params for that eventual flow.

File: src/main/csv.js [hunk 4]
```diff
@@ -1,31 +1,42 @@
...
+	function splitRecursively(parts, index, quoted) {
+		if (index === input.length) {
+			return parts;
+		}
+
+		const char = input.charAt(index++);
+
+		quoted = char === DOUBLE_QUOTE ? !quoted : quoted;
+		if (char !== DOUBLE_QUOTE) {
+			parts = isSeparatorComma(char, quoted) ? startNewBlankItem(parts) : addToLastItem(parts, char);
```
Comment (butlersrepos) [inline src/main/csv.js]: Ternary === Devil. But it makes sense so I just wrapped it in readable terms after I realized what it was doing. Each function is still `functional` and returns a new array built.

File: src/main/csv.js [hunk 5]
```diff
@@ -1,31 +1,42 @@
...
+		}
+
+		return splitRecursively(parts, index, quoted);
+	}
+
+	return splitRecursively([''], 0, false).map(trim);
 }
 
-module.exports = {split: split};
```
Comment (butlersrepos) [inline src/main/csv.js]: don't inline diss
👻 🐍 💩

File: src/test/csvSpec.js [hunk 6]
```diff
@@ -15,9 +15,9 @@ describe('#split', function () {
     });
 
     it('should not split a quoted string', function () {
-        const input = 'He said, "Shall we, my dear?"';
+        const input = 'He said, "Shall we, my dear?", biatch';
         const result = csv.split(input);
-        expect(result).toEqual(['He said', 'Shall we, my dear?']);
+        expect(result).toEqual(['He said', 'Shall we, my dear?', 'biatch']);
```
Comment (butlersrepos) [inline src/test/csvSpec.js]: 99 problems.

File: src/main/csv.js (hunk 1 above)
Comment (aaronmalone) [inline src/main/csv.js]: So you would recommend using `require` on each of those lines, instead of assigning to `_`?

File: src/main/csv.js (hunk 2 above)
Comment (aaronmalone) [inline src/main/csv.js]: Yes.

File: src/main/csv.js (hunk 3 above)
Comment (aaronmalone) [inline src/main/csv.js]: Seems reasonable.

**Pull request merged**